src/cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -838,6 +843,166 @@ static PyObject *log_level(PyObject *, PyObject *args) {
     Py_RETURN_NONE;
 }
 
+class GpiClock {
+  public:
+    GpiClock(GpiObjHdl *clk_sig) : clk_signal(clk_sig) {}
+
+    ~GpiClock() { stop(); }
+
+    int start(uint64_t period_steps, uint64_t high_steps, uint64_t phase_steps);
+
+    int stop();
+
+    bool is_running() const { return clk_toggle_cb_hdl != nullptr; }
+
+  private:
+    GpiObjHdl *clk_signal = nullptr;
+    GpiCbHdl *clk_toggle_cb_hdl = nullptr;
+
+    uint64_t period = 0;
+    uint64_t t_high = 0;
+
+    int clk_val = 0;
+
+    int toggle();
+    static int toggle_cb(void *gpi_clk);
+};
+
+int GpiClock::start(uint64_t period_steps, uint64_t high_steps,
+                    uint64_t phase_steps) {
```

Comment:
I don't really understand this `phase_steps` argument. Why not just pass `start_high` through?